From bbf08fc5aba7a859d4e0ddb96cc00e63edb6e361 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 12 Jan 2026 12:20:38 +0000
Subject: [PATCH 1/1] config: Hardcode profile to look in
 /usr/share/defaults/etc/ instead

We don't need to add statelessness support here as there are other
user mechanisms to override the DEBUGINFOD_URLS.
---
 config/profile.csh.in  | 4 ++--
 config/profile.fish.in | 4 ++--
 config/profile.sh.in   | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/config/profile.csh.in b/config/profile.csh.in
index 1da9626c..cbc78634 100644
--- a/config/profile.csh.in
+++ b/config/profile.csh.in
@@ -6,13 +6,13 @@
 
 set prefix="@prefix@"
 if (! $?DEBUGINFOD_URLS) then
-    set DEBUGINFOD_URLS=`sh -c 'cat /dev/null "$0"/*.urls 2>/dev/null; :' "@sysconfdir@/debuginfod" | tr '\n' ' '`
+    set DEBUGINFOD_URLS=`sh -c 'cat /dev/null "$0"/*.urls 2>/dev/null; :' "/usr/share/defaults/etc/debuginfod" | tr '\n' ' '`
     if ( "$DEBUGINFOD_URLS" != "" ) then
         setenv DEBUGINFOD_URLS "$DEBUGINFOD_URLS"
     else
         unset DEBUGINFOD_URLS
     endif
-    set DEBUGINFOD_IMA_CERT_PATH=`sh -c 'cat /dev/null "$0"/*.certpath 2>/dev/null; :' "@sysconfdir@/debuginfod" | tr '\n' ':'`
+    set DEBUGINFOD_IMA_CERT_PATH=`sh -c 'cat /dev/null "$0"/*.certpath 2>/dev/null; :' "/usr/share/defaults/etc/debuginfod" | tr '\n' ':'`
     if ( "$DEBUGINFOD_IMA_CERT_PATH" != "" ) then
         setenv DEBUGINFOD_IMA_CERT_PATH "$DEBUGINFOD_IMA_CERT_PATH"
     else
diff --git a/config/profile.fish.in b/config/profile.fish.in
index 12b5e6f1..f7b536f7 100644
--- a/config/profile.fish.in
+++ b/config/profile.fish.in
@@ -8,7 +8,7 @@
 set --local prefix "@prefix@"
 
 if not set --query DEBUGINFOD_URLS
-    set --local files "@sysconfdir@/debuginfod/"*.urls
+    set --local files "/usr/share/defaults/etc/debuginfod/"*.urls
     set --local DEBUGINFOD_URLS (cat /dev/null $files 2>/dev/null | string replace '\n' ' ')
     if test -n "$DEBUGINFOD_URLS"
         set --global --export DEBUGINFOD_URLS "$DEBUGINFOD_URLS"
@@ -16,7 +16,7 @@ if not set --query DEBUGINFOD_URLS
 end
 
 if not set --query DEBUGINFOD_IMA_CERT_PATH
-    set --local files "@sysconfdir@/debuginfod/"*.certpath
+    set --local files "/usr/share/defaults/etc/debuginfod/"*.certpath
     set --local DEBUGINFOD_IMA_CERT_PATH (cat /dev/null $files 2>/dev/null | string replace '\n' ':')
     if test -n "$DEBUGINFOD_IMA_CERT_PATH"
         set --global --export DEBUGINFOD_IMA_CERT_PATH "$DEBUGINFOD_IMA_CERT_PATH"
diff --git a/config/profile.sh.in b/config/profile.sh.in
index 9f3e415a..6e477cde 100644
--- a/config/profile.sh.in
+++ b/config/profile.sh.in
@@ -6,12 +6,12 @@
 
 prefix="@prefix@"
 if [ -z "$DEBUGINFOD_URLS" ]; then
-    DEBUGINFOD_URLS=$(find "@sysconfdir@/debuginfod" -name "*.urls" -print0 2>/dev/null | xargs -0 cat 2>/dev/null | tr '\n' ' ' || :)
+    DEBUGINFOD_URLS=$(find "/usr/share/defaults/etc/debuginfod" -name "*.urls" -print0 2>/dev/null | xargs -0 cat 2>/dev/null | tr '\n' ' ' || :)
     [ -n "$DEBUGINFOD_URLS" ] && export DEBUGINFOD_URLS || unset DEBUGINFOD_URLS
 fi
 
 if [ -z "$DEBUGINFOD_IMA_CERT_PATH" ]; then
-    DEBUGINFOD_IMA_CERT_PATH=$(find "@sysconfdir@/debuginfod" -name "*.certpath" -print0 2>/dev/null | xargs -0 cat 2>/dev/null | tr '\n' ':' || :)
+    DEBUGINFOD_IMA_CERT_PATH=$(find "/usr/share/defaults/etc/debuginfod" -name "*.certpath" -print0 2>/dev/null | xargs -0 cat 2>/dev/null | tr '\n' ':' || :)
     [ -n "$DEBUGINFOD_IMA_CERT_PATH" ] && export DEBUGINFOD_IMA_CERT_PATH || unset DEBUGINFOD_IMA_CERT_PATH
 fi
 unset prefix
-- 
2.52.0

